examples/blas/run_time_dispatching/level3/gemm_usm.cpp

```diff
@@ -0,0 +1,247 @@
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+/*
+*
+*  Content:
+*       This example demonstrates use of DPCPP API oneapi::mkl::blas::gemm
+*       using unified shared memory to perform General Matrix-Matrix
+*       Multiplication on a SYCL device (HOST, CPU, GPU) that is selected
+*       during runtime.
+*
+*       C = alpha * op(A) * op(B) + beta * C
+*
+*       where op() is defined by one of oneapi::mkl::transpose::{nontrans,trans,conjtrans}
+*
+*
+*       This example demonstrates only single precision (float) data type for
+*       gemm matrix data
+*
+*
+*******************************************************************************/
+
+// stl includes
+#include <iostream>
+#include <cstdlib>
+#include <limits>
+#include <vector>
+#include <algorithm>
+#include <cstring>
+#include <list>
+#include <iterator>
+
+#include <CL/sycl.hpp>
+#include "oneapi/mkl.hpp"
+
+#include "example_helper.hpp"
+
+//
+// Main example for Gemm consisting of
+// initialization of A, B and C matrices as well as
+// scalars alpha and beta.  Then the product
+//
+// C = alpha * op(A) * op(B) + beta * C
+//
+// is performed and finally the results are post processed.
+//
+void run_gemm_example(const sycl::device& dev) {
+    //
+    // Initialize data for Gemm
+    //
+    // C = alpha * op(A) * op(B)  + beta * C
+    //
+
+    oneapi::mkl::transpose transA = oneapi::mkl::transpose::trans;
+    oneapi::mkl::transpose transB = oneapi::mkl::transpose::nontrans;
+
+    // matrix data sizes
+    int m = 45;
+    int n = 98;
+    int k = 67;
+
+    // leading dimensions of data
+    int ldA = 103;
+    int ldB = 105;
+    int ldC = 106;
+    int sizea = (transA == oneapi::mkl::transpose::nontrans) ? ldA * k : ldA * m;
+    int sizeb = (transB == oneapi::mkl::transpose::nontrans) ? ldB * n : ldB * k;
+    int sizec = ldC * n;
+
+    // set scalar fp values
+    float alpha = set_fp_value(float(2.0), float(-0.5));
+    float beta = set_fp_value(float(3.0), float(-1.5));
+
+    // Catch asynchronous exceptions
+    auto exception_handler = [](sycl::exception_list exceptions) {
+        for (std::exception_ptr const& e : exceptions) {
+            try {
+                std::rethrow_exception(e);
+            }
+            catch (sycl::exception const& e) {
+                std::cerr << "Caught asynchronous SYCL exception during GEMM:" << std::endl;
+                std::cerr << "\t" << e.what() << std::endl;
+            }
+        }
+        std::exit(2);
```

Comment:
Do we need these calls?

Reply:
As commented by Andrey, these std::exit(2) calls allow user to differentiate asynchronous SYCL exceptions from synchronous exceptions (which exits with 1)